trollflow2/tests/test_trollflow2.py

```diff
@@ -2015,5 +2049,88 @@ def fake_persist(*args):
     assert not sc_3a_3b["another"].attrs.get("persisted")
 
 
+yaml_test_s3_uploader_plain = """
+product_list:
+  output_dir: /tmp/
+  publish_topic: /topic
+  s3_config:
+    target: s3://bucket-name/
+  fname_pattern:
+    "{start_time:%Y%m%d_%H%M}_{platform_name}_{areaname}_{productname}.{format}"
+
+  areas:
+    euro4:
+      areaname: euro4
+      products:
+        airmass:
+          productname: airmass
+          formats:
+          - format: tif
+            writer: geotiff
+        natural_with_ir:
+          productname: natural_with_colorized_ir_clouds
+          formats:
+          - format: tif
+            writer: geotiff
+
+"""
+
+
+def test_s3_uploader_update_filenames():
+    """Ensure that filenames are updated when transfer is made to S3."""
+    from yaml import UnsafeLoader
+    from trollflow2.plugins import s3_uploader
+    from trollflow2.dict_tools import plist_iter
+
+    product_list = read_config(raw_string=yaml_test_s3_uploader_plain, Loader=UnsafeLoader)
+    job = {"product_list": product_list, "input_mda": input_mda.copy()}
+    _ = _create_filenames_and_topics(job)
+
+    with mock.patch('trollflow2.plugins.S3Mover'):
+        s3_uploader(job)
+    for fmt, _ in plist_iter(job['product_list']['product_list']):
+        assert fmt['filename'].startswith('s3://bucket-name/')
+
+
+def test_s3_uploader_copy():
+    """Test that S3 mover is only copying the file."""
+    from yaml import UnsafeLoader
+    product_list = read_config(raw_string=yaml_test_s3_uploader_plain, Loader=UnsafeLoader)
+    job = {"product_list": product_list, "input_mda": input_mda.copy()}
+    _ = _create_filenames_and_topics(job)
+
+    with mock.patch('trollflow2.plugins.S3Mover') as S3Mover:
+        from trollflow2.plugins import s3_uploader
+
+        s3_uploader(job)
+
+        assert mock.call(
+            "/tmp/20190217_0600_NOAA-15_euro4_airmass.tif",
+            "s3://bucket-name/"
+        ) in S3Mover.mock_calls
+        assert mock.call(
+            "/tmp/20190217_0600_NOAA-15_euro4_natural_with_colorized_ir_clouds.tif",
+            "s3://bucket-name/"
+        ) in S3Mover.mock_calls
+        assert S3Mover.return_value.copy.call_count == 2
+        S3Mover.return_value.move.assert_not_called()
+
+
+def test_s3_uploader_move():
+    """Test that files are deleted after S3 transfer."""
+    from yaml import UnsafeLoader
+    product_list = read_config(raw_string=yaml_test_s3_uploader_plain, Loader=UnsafeLoader)
+    product_list['product_list']['s3_config']['delete_files'] = True
+    job = {"product_list": product_list, "input_mda": input_mda.copy()}
+    _ = _create_filenames_and_topics(job)
+
+    with mock.patch('trollflow2.plugins.S3Mover') as S3Mover:
+        from trollflow2.plugins import s3_uploader
+
+        s3_uploader(job)
+
+        assert S3Mover.return_value.move.call_count == 2
```

Comment:
Where is this actually checking that files were deleted?

Reply:
The docstring is a bit out-of-sync. The deletion isn't tested directly, just that the `.move()` method of `S3Mover` is called. I'll update the docstring.

Reply:
Updated in https://github.com/pytroll/trollflow2/pull/158/commits/9d103ea3f78bb6341cee3b0dfaf3182b6388a758